spec/services/school_verification_service_spec.rb

```diff
@@ -10,21 +10,62 @@
   let(:organisation_id) { SecureRandom.uuid }
 
   describe '#verify' do
-    before do
-      service.verify
-      school.reload
-    end
+    describe 'when school can be saved' do
+      it 'sets verified_at to a date' do
+        service.verify
+        expect(school.reload.verified_at).to be_a(ActiveSupport::TimeWithZone)
+      end
+
+      it 'grants the creator the owner role for the school' do
+        service.verify
+        expect(school_creator).to be_school_owner(school)
+      end
+
+      it 'grants the creator the teacher role for the school' do
+        service.verify
+        expect(school_creator).to be_school_teacher(school)
+      end
 
-    it 'sets verified_at to a date' do
-      expect(school.verified_at).to be_a(ActiveSupport::TimeWithZone)
+      it 'returns true' do
+        expect(service.verify).to be(true)
+      end
     end
 
-    it 'grants the creator the owner role for the school' do
-      expect(school_creator).to be_school_owner(school)
+    describe 'when school cannot be saved' do
+      before do
+        school.update_attribute(:website, 'invalid') # rubocop:disable Rails/SkipsModelValidations
```

Comment:
Minor: It would be nicer to use a `let` blocks to avoid having to update the school in the database and avoid having to disable the rubocop rule.

Reply:
I've addressed this in a new commit titled "Avoid update_attribute in SchoolVerificationService spec".